raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -11,11 +11,12 @@ contract ServiceRegistry is Utils {
     mapping(address => string) public urls;  // URLs of services for HTTP access
     address[] public service_addresses;  // list of available services (ethereum addresses)
 
-    constructor(address _token_address) public {
-        require(_token_address != address(0x0));
-        require(contractExists(_token_address));
+    // @param _token_for_registration The address of the ERC20 token contract that services use for registration fees
+    constructor(address _token_for_registration) public {
+        require(_token_for_registration != address(0x0));
+        require(contractExists(_token_for_registration));
 
-        token = Token(_token_address);
```

Comment:
Why is the code duplicated? Both here and in ` raiden_contracts/contracts/services/ServiceRegistry.sol` ?

Reply:
Just for historical reasons.  I guess `make compile_contracts` and so should just look at `data/source/...`, and the other can be removed.

Reply:
https://github.com/raiden-network/raiden-contracts/issues/740

Reply:
Yes please let's do that, maintaining duplicated contract code sounds like a nightmare.